test/mpi/rma/lockall_dt_flushlocal.c

```diff
@@ -34,82 +58,114 @@ int main(int argc, char *argv[])
+
+            recvcount = recv_dtp->DTP_obj_array[i].DTP_obj_count;
+            recvtype = recv_dtp->DTP_obj_array[i].DTP_obj_type;
+            recvbuf = recv_dtp->DTP_obj_array[i].DTP_obj_buf;
+
+            MPI_Type_get_extent(recvtype, &lb, &extent);
+
+            MPI_Win_create(recvbuf, lb + recvcount * extent,
+                           (int) extent, MPI_INFO_NULL, comm, &win);
+
+            for (j = 0; j < send_dtp->DTP_num_objs; j++) {
+                err = DTP_obj_create(send_dtp, j, 0, 1, count);
+                if (err != DTP_SUCCESS) {
+                    MPI_Type_get_name(basic_type, type_name, &len);
+                    fprintf(stdout, "Error while creating obj %d in send pool (%s,%d)\n", j,
+                            type_name, count);
+                    fflush(stdout);
+                    break;
+                }
+
+                sendcount = send_dtp->DTP_obj_array[j].DTP_obj_count;
+                sendtype = send_dtp->DTP_obj_array[j].DTP_obj_type;
+                sendbuf = send_dtp->DTP_obj_array[j].DTP_obj_buf;
 
-                MPI_Win_create(recvtype.buf, lb + recvtype.count * extent,
-                               (int) extent, MPI_INFO_NULL, comm, &win);
                 if (rank == source) {
                     int dest;
                     MPI_Aint slb, sextent;
-                    MPI_Type_get_extent(sendtype.datatype, &slb, &sextent);
-                    sendtype.InitBuf(&sendtype);
+                    MPI_Type_get_extent(sendtype, &slb, &sextent);
+
+                    char *tmp = (char *) malloc(slb + sextent * sendcount);
+                    memcpy(tmp, sendbuf, slb + sextent * sendcount);
 
                     MPI_Win_lock_all(0, win);
 
                     for (dest = 0; dest < size; dest++)
                         if (dest != source) {
-                            MPI_Accumulate(sendtype.buf, sendtype.count,
-                                           sendtype.datatype, dest, 0,
-                                           recvtype.count, recvtype.datatype, MPI_REPLACE, win);
+                            MPI_Accumulate(sendbuf, sendcount,
+                                           sendtype, dest, 0,
+                                           recvcount, recvtype, MPI_REPLACE, win);
                             MPI_Win_flush_local(dest, win);
                         }
                     /* reset the send buffer to test local completion */
-                    memset(sendtype.buf, 0, slb + sextent * sendtype.count);
+                    memset(sendbuf, 0, slb + sextent * sendcount);
                     MPI_Win_unlock_all(win);
                     MPI_Barrier(comm);
 
-                    sendtype.InitBuf(&sendtype);
-                    char *resbuf = (char *) calloc(lb + extent * recvtype.count, sizeof(char));
+                    memcpy(sendbuf, tmp, slb + sextent * sendcount);
+                    char *resbuf = (char *) calloc(lb + extent * recvcount, sizeof(char));
 
```

Comment:
A few years ago, we had discussed that the `resbuf` should be checked as well. But with the old mtest routine, we could not make it. This plan should work with dtpool now. Do you think you can add it in as separate commit ? It is OK to leave it as a TODO.